File: www/cgi-bin/test_oracle.py
#!/usr/bin/env python3
import cgitb
cgitb.enable()
import os
import cx_Oracle

def main():
  l_database = 'OPVDA'
  print('Content-Type: text/html;charset=utf-8')
  print()
  print("""<html>
 <head>
  <title> Test met python en oracle </title>
 </head>
 <body>
""")
  con = cx_Oracle.connect('/@'+l_database)

  cur = con.cursor()
  t="select user from dual"
  cur.execute(t)
  results = []
  for result in cur:
    results.append(result)
    print("<h1>" + result[0] + "</h1>")  
  
  #print(results)
  cur.close()
  con.close()

  print("""</body>
</html>
""")

if __name__ == '__main__':
  main()

File: www/cgi-bin/test_oracle.rb
#!/usr/bin/env ruby
require 'oci8'
###########
## Main
###########

puts 'Content-Type: text/html;charset=utf-8'
puts ""
puts <<-EOS
<html>
 <head>
  <title> Test met ruby en oracle </title>
 </head>
 <body>
EOS
dbs = 'OPVDA'
conn = OCI8.new(nil,nil,dbs)
cur = conn.parse('select user from dual')
cur.exec
cur.fetch do |v|
  printf "<h1>%s</h1>\n", v[0]
end
cur.close
conn.logoff
puts <<-EOS
</body>
</html>
EOS

